patches/server/0975-Add-speed-argument-to-ProjectileSource-launchProject.patch

```diff
@@ -0,0 +1,153 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: wyrdix <wyrdix@github.com>
+Date: Sat, 22 Apr 2023 17:39:23 +0200
+Subject: [PATCH] Add speed argument to ProjectileSource#launchProjectile
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+index d43859f8aa7beed82dd3a146bb1086982cd0cda7..e5eb4cdb69f761709ecf8a45183ba50f472bef8c 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+@@ -83,8 +83,11 @@ import org.bukkit.potion.PotionEffect;
+ import org.bukkit.potion.PotionEffectType;
+ import org.bukkit.potion.PotionType;
+ import org.bukkit.util.BlockIterator;
++import org.bukkit.util.Consumer;
+ import org.bukkit.util.RayTraceResult;
+ import org.bukkit.util.Vector;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
+ 
+ public class CraftLivingEntity extends CraftEntity implements LivingEntity {
+     private CraftEntityEquipment equipment;
+@@ -502,13 +505,25 @@ public class CraftLivingEntity extends CraftEntity implements LivingEntity {
+     @Override
+     public <T extends Projectile> T launchProjectile(Class<? extends T> projectile, Vector velocity) {
+         // Paper start - launchProjectile consumer
+-        return this.launchProjectile(projectile, velocity, null);
++        return this.launchProjectile(projectile, velocity, 1f,null);
++    }
++
++    @Override
++    public <T extends Projectile> @NotNull T launchProjectile(@NotNull Class<? extends T> projectile, @Nullable Vector velocity, @Nullable Consumer<T> function) {
++        return this.launchProjectile(projectile, velocity, 1f, function);
++    }
++
++    @Override
++    public <T extends Projectile> @NotNull T launchProjectile(@NotNull Class<? extends T> projectile, @Nullable Vector velocity, float speed) {
++        // Paper start - initial speed
++        return this.launchProjectile(projectile, velocity, speed, null);
+     }
+ 
+     @Override
+     @SuppressWarnings("unchecked")
+-    public <T extends Projectile> T launchProjectile(Class<? extends T> projectile, Vector velocity, org.bukkit.util.Consumer<T> function) {
++    public <T extends Projectile> T launchProjectile(Class<? extends T> projectile, Vector velocity, float speed, org.bukkit.util.Consumer<T> function) {
+         // Paper end - launchProjectile consumer
++        // Paper end - initial speed
```

Comment:
two ends?

Reply:
I didn't know how to manage this, I tried to fit the contributing rules as much as possible.
I have another start above so it seemed right to put another end